packages/nimble-components/src/select/index.ts

```diff
@@ -1,31 +1,83 @@
-import { attr, html, observable } from '@microsoft/fast-element';
+// eslint-disable-next-line max-classes-per-file
+import {
+    attr,
+    DOM,
+    html,
+    observable,
+    Observable,
+    volatile
+} from '@microsoft/fast-element';
 import {
     AnchoredRegion,
     DesignSystem,
     Select as FoundationSelect,
-    SelectOptions
+    ListboxOption,
+    SelectOptions,
+    SelectPosition,
+    applyMixins,
+    StartEnd,
+    DelegatesARIASelect,
+    Listbox
 } from '@microsoft/fast-foundation';
+import {
+    keyArrowDown,
+    keyArrowUp,
+    keyEnd,
+    keyEnter,
+    keyEscape,
+    keyHome,
+    keySpace,
+    keyTab,
+    uniqueId
+} from '@microsoft/fast-web-utilities';
 import { arrowExpanderDown16X16 } from '@ni/nimble-tokens/dist/icons/js';
 import { styles } from './styles';
 import { DropdownAppearance } from '../patterns/dropdown/types';
 import { errorTextTemplate } from '../patterns/error/template';
 import type { ErrorPattern } from '../patterns/error/types';
 import { iconExclamationMarkTag } from '../icons/exclamation-mark';
 import { template } from './template';
+import type { ListOption } from '../list-option';
+import { FilterMode } from './types';
+import { diacriticInsensitiveStringNormalizer } from '../utilities/models/string-normalizers';
+import { FormAssociatedSelect } from './models/select-form-associated';
 
 declare global {
     interface HTMLElementTagNameMap {
         'nimble-select': Select;
     }
 }
 
+// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+type BooleanOrVoid = boolean | void;
+
 /**
- * A nimble-styled HTML select
+ * A nimble-styled HTML select. The FAST Select implementation has largely been
```

Comment:
nit: this commentary about FAST shouldn't appear in the class JSDocs as they are client-facing but that info is an implementation detail. You could follow the pattern we've used in other forks of putting a comment at the top of the file or somewhere else that's not public and JSDoc syntax.